lib/select.c

```diff
@@ -534,10 +535,8 @@ CURLcode Curl_waitfds_add_ps(struct curl_waitfds *cwfds,
       events |= CURL_WAIT_POLLIN;
     if(ps->actions[i] & CURL_POLL_OUT)
       events |= CURL_WAIT_POLLOUT;
-    if(events) {
-      if(cwfds_add_sock(cwfds, ps->sockets[i], events))
-        return CURLE_OUT_OF_MEMORY;
-    }
```

Comment:
This is an error detection that properly returns an error that you remove here. It does not seem okay, We should not ignore error, that will lead to weird and inexplicable behaviors.

Reply:
This error condition is covered, but how it works has changed.

The functions `cwfds_add_sock`, `Curl_waitfds_add_ps`, and `Curl_cpool_add_waitfds` were changed to no longer return an error, but rather return the  number of entries they needed. These functions will still ensure they do not overwrite past the end of the buffer.

The out of memory test has been moved to the following block.

```
  if(need != cwfds.n && ufds) {
    result = CURLM_OUT_OF_MEMORY;
  }
```

Notably if the need is not equal to the number of entries written out, then an `CURLM_OUT_OF_MEMORY`. It should never be the case that `need < cwfds.n`

Another option, is to retain the error code, and add a output parameter to all three functions that updates the `need`. I tried both methods when developing this pull request, and this one seemed simpler.